dpnp/dpnp_utils/dpnp_utils_statistics.py

```diff
@@ -69,6 +74,10 @@ def _get_2dmin_array(x, dtype):
             x = x.T
 
         if x.dtype != dtype:
+            fp64 = x.sycl_device.has_aspect_fp64
+            # TODO: remove when dpctl.result_type() is fixed
+            if not fp64:
+                dtype = dtype_map[dtype.type]
```

Comment:
I'd propose to move the w/a to `if` block below where `dpt.result_type` is called, otherwise you are hiding the issue when incoming `dtype=dpnp.float64` is passed to `dpnp.cov()`.